flatflow/data/dataset_test.cc

```diff
@@ -92,4 +99,112 @@ TEST(DatasetTest, Constructor) {
   EXPECT_TRUE(dataset.empty());
 }
 
+TEST(DatasetTest, IntraBatchShuffling) {
+  // Test case for IntraBatchShuffling
+  //
+  // NOTE:
+  //
+  // * We first check if the dataset vectors are sorted.
+  // * After shuffling, we check if dataset vectors are not sorted.
+  // * Then using the same input vector, we shuffle the slots.
+  // * Last, we check if the slots and dataset vectors are equal. 
+
+  std::srand(static_cast<unsigned int>(std::time(nullptr)));
+  auto items = std::map<uint16_t, std::size_t>();
+  uint16_t epoch = 0;
+
+  for (uint16_t size = 1; size <= 1 << 12; ++size) {
+    items.emplace(size, static_cast<std::size_t>(std::rand() % (1 << 15)));
+  }
+
+  auto sizes = std::vector<uint16_t>();
+  for (const auto item : items) {
+    const auto size = item.first;
+    auto count = item.second;
+    for (; 0 < count; --count) {
+      sizes.push_back(size);
+    }
+  }
+  sizes.shrink_to_fit();
+
+  auto builder = flatbuffers::FlatBufferBuilder64();
+  auto sizes__ = builder.CreateVector64(sizes);
+  auto offset = CreateSizes(builder, sizes__);
+  builder.Finish(offset);
+
+  auto sizes_ = GetSizes(builder.GetBufferPointer());
+  auto dataset = DatasetTest(sizes_->sizes(), 0UL);
+
+  // Expects all vectors are sorted.
+  for (const auto &item : items) {
```

Comment:
The constructor test already verifies that each index slot is sorted upon construction of inverted index, so this test can be removed.